File: 234-PalindromeLinkedList/234-PalindromeLinkedList.py
# Last updated: 12/6/2025, 5:51:15 am
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        left = head
        def rec(node):
            nonlocal left
            if node == None: return True
            if not rec(node.next) or node.val != left.val: return False
            left = left.next
            return True
        return rec(head)

File: 234-PalindromeLinkedList/234-PalindromeLinkedList.js
// Last updated: 12/6/2025, 5:51:26 am
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
    const rec = (node) => {
        if (!node) return 1;
        else if (!rec(node.next) || node.val != head.val) return 0;
        head = head.next;
        return 1;
    }
    return rec(head);
};